admin_manual/installation/nginx.rst

```diff
@@ -140,15 +140,15 @@ webroot of your nginx installation. In this example it is
       }
       
       # Rules borrowed from `.htaccess` to hide certain paths from clients
-      location ~ ^/(build|tests|config|lib|3rdparty|templates|data)($|/)  { return 404; }
-      location ~ ^/(\.|autotest|occ|issue|indie|db_|console)              { return 404; }
+      location ~ ^/(?:build|tests|config|lib|3rdparty|templates|data)($|/)  { return 404; }
+      location ~ ^/(?:\.|autotest|occ|issue|indie|db_|console)              { return 404; }
       
       # Ensure this block, which passes PHP files to the PHP process, is above the blocks
       # which handle static assets (as seen below). If this block is not declared first,
       # then Nginx will encounter an infinite rewriting loop when it prepends `/index.php`
       # to the URI, resulting in a HTTP 500 error response.
       location ~ \.php($|/) {
-          fastcgi_split_path_info ^(.+?\.php)(/.*|)$;
+          fastcgi_split_path_info ^(.+?\.php)(/.*)$;
```

Comment:
Are we sure that `(/.*)` rather than `(/.*|)` correctly sets `$fastcgi_path_info` for a URI such as `index.php` (no trailing slash), wherein `(/.*)` doesn't match anything?

Reply:
This appears on lines 146 and 316.

Reply:
I am fairly certain (but feel free to test). This is how the configuration used to be, until it was changed without a proper discussion [here](https://github.com/nextcloud/documentation/pull/1152).

In the case of `index.php`, `fastcgi_split_path_info` will not match and `$fastcgi_path_info` will not be set, which seems like the appropriate behavior to me.

Reply:
If you're ok with my change/explanation here then it looks like everything is resolved and I am all good giving a "thumbs up"

Reply:
I will be doing final tests now on a fresh install just to check this and whether there's any obvious issues.

Reply:
https://nextcloud.com/blog/urgent-security-issue-in-nginx-php-fpm/

Reply:
> https://nextcloud.com/blog/urgent-security-issue-in-nginx-php-fpm/

Nevermind. I mixed it up.

Reply:
> I will be doing final tests now on a fresh install just to check this and whether there's any obvious issues.

Ok great. I'm modified my production configuration and am basically using what we have here, so there's that datapoint.